### ISSUE TITLE
_tmp_<hexhash>_<filename>.rb files litter my project

### ISSUE THREAD

**Issue by Nitrodist** [castwide/solargraph#341]
Hello, I use solargraph with nvim + coc.nvim. I recently saw new files popping up in my project:

![image](https://user-images.githubusercontent.com/385448/87689253-9be35e80-c74d-11ea-9294-7082809755bf.png)

Here's the code in the solargraph gem that I think is generating it:

```
lib/solargraph/language_server/message/text_document/formatting.rb:            tempfile = File.join(File.dirname(filename), "_tmp_#{SecureRandom.hex(8)}_#{File.basename(filename)}.rb")
```

I suggest moving the File.unlink to an `ensure` block.

**Comment by Nitrodist** [comment]
See https://github.com/castwide/solargraph/pull/342 for a PR fix.

Here's another user report: https://www.reddit.com/r/neovim/comments/gmq965/vim_creating_tmp_files/

**Comment by castwide** [comment]
Thanks! Released in v0.39.12.
